src/core/Tw2Vector4Parameter.js

```diff
@@ -99,4 +99,14 @@ Tw2Vector4Parameter.prototype.SetIndexValue = function(index, value)
     }
 };
 
+Tw2Vector4Parameter.prototype.FillWith = function(number)
+{
+    if (/^-?[\d.]+(?:e-?\d+)?$/.test(number))
+    {
+        this.SetValue(number, number, number, number);
```

Comment:
FillWith is explicit, and there isn't any type checking in place in the ccpwgl code you could accidentally do a fill with the SetValue() when you didn't mean to.